firmware/arduino/chessboard/uart.ino

```diff
@@ -58,54 +67,91 @@ void serialEvent2()
             byteNum = -1;
 
             // Process input
-            // Returns true for valid input and false for invalid input. Calls movement function
+            // Returns true for valid input and false for invalid input.
             currentState = EXECUTING;
-            if (parseMessageFromPi(buffer))
-                errorCode = NONE;
+            if (validateMessageFromPi(buffer))
+                errorCode = NO_ERROR;
+
 
-            // Send response
+            // Sends acknowledgement and any errors with the instruction
             sendMessageToPi(currentState, buffer[5], errorCode);
+
+            if (errorCode == NO_ERROR)
+            {
+                // Makes move, returns success
+                if(makeMove(buffer))
+                    currentState = IDLE;
+
+                sendMessageToPi(currentState, buffer[5], errorCode);
+            }
         }
     }
 }
 
-bool parseMessageFromPi(char * message)
+// Check that the instruction is valid
+bool validateMessageFromPi(char * message)
 {
-    // Move types 0 and 1 are valid for the same range of values
-    if (message[0]  ==  DIRECT  ||  message[0]  ==  EDGES)
+    if (message[0] == DIRECT || message[0] == EDGES)
     {
-        // Check that the inputs are in a valid range
-        if (message[1] < 'A'  ||  message[1] > 'L' 
-           ||  message[2] < 'A'  ||  message[2] > 'L' 
-           ||  message[3] < 'A'  ||  message[3] > 'L' 
-           ||  message[4] < 'A'  ||   message[4] > 'L') 
+        if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]) 
+            || isInvalidCoord(message[3]) || isInvalidCoord(message[4]))
```

Comment:
Pls align the text in the second line with the first condition above, makes it easier to read.
```c
if (isInvalidCoord(message[1]) || isInvalidCoord(message[2]) ||
    isInvalidCoord(message[3]) || isInvalidCoord(message[4]))
```